pyperformance/data-files/benchmarks/bm_async_tree/run_benchmark.py

```diff
@@ -47,17 +48,31 @@ async def workload_func(self):
             "To be implemented by each variant's derived class."
         )
 
-    async def recurse(self, recurse_level):
+    async def recurse_with_gather(self, recurse_level):
         if recurse_level == 0:
             await self.workload_func()
             return
 
         await asyncio.gather(
-            *[self.recurse(recurse_level - 1) for _ in range(NUM_RECURSE_BRANCHES)]
+            *[self.recurse_with_gather(recurse_level - 1)
+              for _ in range(NUM_RECURSE_BRANCHES)]
+        )
+
+    async def recurse_with_task_groups(self, recurse_level):
+        if recurse_level == 0:
+            await self.workload_func()
+            return
+
+        await asyncio.gather(
```

Comment:
This should use taskgroup right not gather?

Reply:
haha, of course, good catch! I used gather in both for local testing with 3.10 and forgot to change back 😬